[quant][graphmode] Support conv*d_relu in traced models

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39615 [quant][graphmode] Fix a corner case in handling `if` in insert_observers
* #39551 [quant][graphmode] Support propagate dequantize for nodes with multiple outputs
* #39550 [quant][graphmode][refactor] propagateQuantizationOps
* **#39490 [quant][graphmode] Support conv*d_relu in traced models**
* #39455 [quant][graphmode] Fix add_relu patterns for scripting and tracing

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21917117](https://our.internmc.facebook.com/intern/diff/D21917117)

jerryzh168 committed:
[quant][graphmode] Support conv*d_relu in traced models

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Support conv*d_relu in traced models"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Support conv*d_relu in traced models"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Support conv*d_relu in traced models"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Support conv*d_relu in traced models"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Support conv*d_relu in traced models"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Support conv*d_relu in traced models"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank100 commented on torch/csrc/jit/passes/quantization/insert_observers.cpp:
```diff
@@ -410,54 +410,101 @@ class InsertObserversHelper {
+  const PatternInfo nn_conv2d_nn_relu = PatternInfo::parse_from_str(
+      R"(
+graph(%self, %input, %relu):
+    %first_module = match::module[name="Conv2d"](%self)
```
Why is conv2d the exception here?

raghuramank100 reviewed (approved):
Looks good, except for special casing of Conv2d.

jerryzh168 committed:
Update on "[quant][graphmode] Support conv*d_relu in traced models"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21917117](https://our.internmc.facebook.com/intern/diff/D21917117)

[ghstack-poisoned]
